services/venice-server/src/main/java/com/linkedin/venice/listener/StorageReadRequestHandler.java

```diff
@@ -349,6 +353,68 @@ public void channelRead(ChannelHandlerContext context, Object message) throws Ex
     }
   }
 
+  @Override
+  public void grpcRead(GrpcHandlerContext ctx, GrpcHandlerPipeline pipeline) {
```

Comment:
Is it the right understanding that this code executes the request inline within gRPC's (Netty?) worker thread? In the old code, we delegate the request execution to another thread pool. I wonder if we should do the same in gRPC (and if it's even possible at all).

Reply:
That is correct, it will be executed with a gRPC worker thread. We can define the executor that gRPC uses to better suit our needs, but at the same time we should be able to reuse or define another thread pool to handle storage requests for grpc.